src/types.hpp

```diff
@@ -7,9 +7,12 @@
 
 // Type declarations for node ids and way ids.  We don't need the full 64 bits
 // for ways as the highest way number isn't close to 2^32 yet.
-typedef std::uint64_t external_nodeid_t;
+typedef std::int64_t external_nodeid_t;
```

Comment:
Yep check
https://github.com/nodejs/nan/blob/master/doc/converters.md#api_nan_to
I would use the `To<int64_t>` function, assert for `> 0` and `static_cast` to your unsigned type then. Negative ids do not make sense after all - the Js->C++ code should assert that.